bootstrap.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env python
+# coding=utf-8
+import argparse
+import os
+import subprocess
+from string import Template
+
+from blessings import Terminal
+
+
+DEFAULT_DIRECTORY_NAME = 'mozilla-payments'
+GIT_REPOS = [
+    {'org': 'Kinto', 'name': 'kinto'},
+    {'org': 'mozilla', 'name': 'payments-example'},
+    {'org': 'mozilla', 'name': 'payments-service'},
+    {'org': 'mozilla', 'name': 'payments-ui'},
+    {'org': 'mozilla', 'name': 'solitude'},
+    {'org': 'mozilla', 'name': 'solitude-auth'},
+]
+PAYMENT_ENV_DIR = os.path.abspath(os.path.dirname(__file__))
+LOCAL_ENV = Template((os.fdopen(os.open(
+    os.path.join(PAYMENT_ENV_DIR, 'env', 'local.env.dist'),
+    os.O_RDWR), "r+").read()))
+SEPARATOR = """
+###############################################################################
+💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰 💰
+###############################################################################"""
+
+class ShellCommandError(Exception):
+    pass
+
+
+def main():
+    args = get_arguments()
+
+    term = Terminal()
+
+    print (
+"""{s}
+
+Hi there! 😀
+
+Time to set up your 💰 {t.red}{t.bold}Mozilla Payments{t.normal}💰  environment!
+
+First, we are going to create a personlised environment file so you can
+process payments in development mode. You'll need to have your
+{t.bold}Braintree API keys{t.normal}.
+
+If you don't have a Braintree account yet or don't know your API Keys,
+go to {t.blue_underline}braintree.com{t.normal} and create an account/log in, then go to
+{t.blue_underline}solitude.readthedocs.org/latest/topics/setup.html#braintree-settings{t.normal}
+to see how to get your API keys.
+""").format(
+        dir=os.path.abspath(os.path.join(PAYMENT_ENV_DIR, '..', args.dir)),
+        s=SEPARATOR, t=term
+    )
+
+    env_vars = get_environment_variables()
+
+    create_local_env(env_vars)
+
+    print (
+"""{s}
+
+Okay, that's all finished. 👍
+
+Next we're going to check out the {t.red}{t.bold}Mozilla Payments{t.normal} git repositories to:
+
+{dir}
+""").format(dir=os.path.abspath(os.path.join(PAYMENT_ENV_DIR, '..', args.dir)),
+            s=SEPARATOR, t=term)
+
+    # Create our payments repos.
+    create_folder_and_checkout_repos(args.dir)
+
+    print(
+"""{s}
+
+Repositories downloaded! ✅
+
+Next we're going to download Docker images and data.
+This will mean hundreds of MBs of downloads and will likely take some time.
+
+Make sure you have a good internet connection. Maybe get a coffee 😉
+""".format(s=SEPARATOR))
+    bootstrap_docker()
+
+    print(
+"""{s}
+
+Docker all set up! ✅
+
+Payments UI needs some NPM/bower modules to run; we'll install them now.
+""".format(s=SEPARATOR))
+
+    bootstrap_ui(args.dir)
+
+    print (
+"""{s}
+
+Setup complete. 👏
+
+For now there are a few manual, one-time steps.
+
+1. Run this command:
+
+    echo "$(docker-machine ip default) pay.dev" | sudo tee -a /etc/hosts
```

Comment:
Oh, so the machine might not be running even though I've run docker-compose up -d? Darn.




I'll see about fixing that race condition and doing the pep8 bit.



- tofumatt

On Sat, Sep 19, 2015 at 12:50 AM, Andy McKay <notifications@github.com>
wrote:

>> +
>> +Payments UI needs some NPM/bower modules to run; we'll install them now.
>> +""".format(s=SEPARATOR))
>> +
>> +    bootstrap_ui(args.dir)
>> +
>> +    print (
>> +"""{s}
>> +
>> +Setup complete. 👏
>> +
>> +For now there are a few manual, one-time steps.
>> +
>> +1. Run this command:
>> +
>> +    echo "$(docker-machine ip default) pay.dev" | sudo tee -a /etc/hosts
> Handling docker-machine is a pita in the ass though, not sure what we can do there.
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/mozilla/payments-env/pull/55/files#r39911693